exasol_transformers_extension/utils/model_downloader.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+from typing import Protocol, runtime_checkable
+
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+
+from exasol_transformers_extension.utils.bucketfs_model_uploader import BucketFSModelUploaderFactory
+from exasol_transformers_extension.utils.temporary_directory_factory import TemporaryDirectoryFactory
+
+
+@runtime_checkable
+class ModelFactoryProtocol(Protocol):
+    def from_pretrained(self, model_name: str, cache_dir: Path, use_auth_token: str):
```

Comment:
what does this do?

Reply:
Protocols are for static duck typing. Here some resources:
- https://www.daan.fyi/writings/python-protocols
- https://en.wikipedia.org/wiki/Duck_typing